x-pack/test/cases_api_integration/common/plugins/cases/server/routes.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createHash } from 'crypto';
+import semverCompare from 'semver/functions/compare';
+import type { CasesPatchRequest } from '@kbn/cases-plugin/common/api';
+import { schema } from '@kbn/config-schema';
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type {
+  ExternalReferenceAttachmentType,
+  PersistableStateAttachmentTypeSetup,
+} from '@kbn/cases-plugin/server/attachment_framework/types';
+import type { FixtureStartDeps } from './plugin';
+
+const hashParts = (parts: string[]): string => {
+  const hash = createHash('sha1');
+  const hashFeed = parts.join('-');
+  return hash.update(hashFeed).digest('hex');
+};
+
+const extractMigrationInfo = (type: PersistableStateAttachmentTypeSetup) => {
+  const migrationMap = typeof type.migrations === 'function' ? type.migrations() : type.migrations;
+  const migrationVersions = Object.keys(migrationMap ?? {});
+  migrationVersions.sort(semverCompare);
+
+  return { migrationVersions };
+};
+
+const getExternalReferenceAttachmentTypeHash = (type: ExternalReferenceAttachmentType) => {
+  return hashParts([type.id]);
+};
+
+const getPersistableStateAttachmentTypeHash = (type: PersistableStateAttachmentTypeSetup) => {
+  const { migrationVersions } = extractMigrationInfo(type);
+
+  return hashParts([type.id, migrationVersions.join(',')]);
```

Comment:
If a plugin registers a persistable attachment type and creates a new migration that will trigger this to cause a failure right?

Reply:
Correct!